x-pack/platform/plugins/shared/alerting/server/alerts_client/lib/inject_analyze_wildcard.ts

```diff
@@ -7,24 +7,37 @@
 
 import type { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
 
+const MAX_QUERIES = 20000;
+
 export const injectAnalyzeWildcard = (query: QueryDslQueryContainer): void => {
   if (!query) {
     return;
   }
 
-  if (Array.isArray(query)) {
-    return query.forEach((child) => injectAnalyzeWildcard(child));
-  }
+  let queriesCount = 0;
+  const stack: QueryDslQueryContainer[] = [query];
 
-  if (typeof query === 'object') {
-    Object.entries(query).forEach(([key, value]) => {
-      if (key !== 'query_string') {
-        return injectAnalyzeWildcard(value);
-      }
+  while (stack.length > 0) {
+    queriesCount = queriesCount + 1;
 
-      if (typeof value.query === 'string' && value.query.includes('*')) {
-        value.analyze_wildcard = true;
+    if (queriesCount > MAX_QUERIES) {
```

Comment:
We should test whether this error is thrown in `x-pack/platform/plugins/shared/alerting/server/alerts_client/lib/inject_analyze_wildcard.test.ts` by mocking `MAX_QUERIES`.